test/remote_docker.py

```diff
@@ -0,0 +1,52 @@
+import os
+import argparse
+from subprocess import check_output, check_call
+import socket
+import random
+
+def detect_port(port):
+    '''Detect if the port is used, return True if the port is used'''
+    socket_test = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    try:
+        socket_test.connect(('127.0.0.1', int(port)))
+        socket_test.close()
+        return True
+    except:
+        return False
+
+def find_port():
+    '''Find a port which is free'''
+    port = random.randint(5000, 10000)
+    while detect_port(port):
+        port = random.randint(5000, 10000)
+    return port
+
+def start_container(image, name):
+    '''Start docker container'''
+    port = find_port()
+    source_dir = '/tmp/nnitest/' + name
+    run_cmds = ['docker', 'run', '-d', '-p', str(port) + ':22', '--name', name, '--mount', 'type=bind,source=' + source_dir + ',target=/tmp/nni', image]
+    output = check_output(run_cmds)
+    commit_id = output.decode('utf-8')
+    exec_cmds = ['docker', 'exec', name, 'python3', '-m', 'pip', 'install', '--user', '--no-cache-dir', '/tmp/nni/pynni/']
+    check_call(exec_cmds)
+    with open(source_dir + '/port', 'w') as file:
+        file.write('    port: ' + str(port))
```

Comment:
Suggest write only the port value to file,  then format the string 'port:  xxx' from outside.